staging/src/k8s.io/apiserver/pkg/util/webhook/webhook_test.go

```diff
@@ -867,7 +867,7 @@ func TestWithExponentialBackoffParametersNotSet(t *testing.T) {
 
 	err := WithExponentialBackoff(context.TODO(), wait.Backoff{}, webhookFunc, alwaysRetry)
 
-	errExpected := fmt.Errorf("webhook call failed: %s", wait.ErrWaitTimeout)
+	errExpected := fmt.Errorf("webhook call failed: %s", wait.ErrorInterrupted(nil))
```

Comment:
I think the `WithExponentialBackoff` should return a wrapped error. and then we can use `wait.Interrupted(err)` to check it.

i.e.

```go
	switch {
	// we check for webhookErr first, if webhookErr is set it's the most important error to return.
	case webhookErr != nil:
		return webhookErr
	case err != nil:
		return fmt.Errorf("webhook call failed: %w", err.Error())
	default:
		return nil
	}
```


https://github.com/kubernetes/kubernetes/blob/master/test/integration/apimachinery/watch_restart_test.go#L241-L262